File: .net & mono/Weed3Demo/demo_plus/queryProcedure/user_update2.cs
using Noear.Weed;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

//如果原来是存储过程的代码，可以通过[DbQueryProcedure]快速切换过来

namespace Weed3Demo.demo_plus.queryProcedure {
    class user_update2 : DbQueryProcedure {
        public user_update2(): base(DbConfig.test) {

            sql("update user set city=@city,vipTime=@vipTime where userID=@userID");
            set("@userID", ()=>userID);
            set("@city", ()=>city);
            set("@vipTime", ()=>vipTime);
        }

        public long userID;
        public String city;
        public DateTime vipTime;
    }

}

File: _test/src/main/java/weed3demo/demo_plus/queryProcedure/user_update2.java
package weed3demo.demo_plus.queryProcedure;

import org.noear.weed.DbQueryProcedure;
import org.noear.weed.DbStoredProcedure;
import weed3demo.config.DbConfig;

import java.util.Date;

/**
 * Created by noear on 2017/7/22.
 */

//如果原来是存储过程的代码，可以通过[DbQueryProcedure]快速切换过来

public class user_update2 extends DbQueryProcedure {
    public user_update2() {
        super(DbConfig.test);

        lazyload(()->{
            sql("update user set city=@city,vipTime=@vipTime where userID=@userID");
            set("userID",  userID);
            set("city",  city);
            set("vipTime",  vipTime);
        });
    }

    public long userID;
    public String city;
    public Date vipTime;
}


